addons/pos_self_order/views/res_config_settings_views.xml

```diff
@@ -54,8 +54,26 @@
                         <field name="pos_self_ordering_image_brand_name" invisible ="1"/>
                         <field name="pos_self_ordering_image_brand" class="w-100" filename="pos_self_ordering_image_brand_name"/>
                     </setting>
-                    <setting string="Allow takeout order" help="Allow self-order customers to set their order for takeout." invisible="not pos_self_ordering_mode in ['mobile', 'kiosk'] or not pos_takeaway">
+                    <setting string="Allow takeout order" help="Allow self-order customers to set their order for takeout." invisible="not pos_self_ordering_mode in ['mobile', 'kiosk'] or not pos_use_presets">
```

Comment:
Check if pos_use_preset is needed. The thing is that, if you do not check the pos_preset option, then select your point of sale to be a kiosk, you lose the access to pos_use_presets and thus the Allow takeout order option will not show. If the pos_use_presets is not needed, then you will have the options and the user will still be able to chose it or not